### PR TITLE
Add autocomplete dropdown and improved search

### PR DESCRIPTION
**Pull request by maysakanoni**
**What**  
This pr introduces a dropdown in the search which dynamically updates with the search. It pulls all the search terms from air-table

**Why**  
This will allow users to anticipate the search terms

image

### CODE REVIEW

File: components/Feature/TopicExplorer/index.js [hunk 1]
```diff
@@ -1,25 +1,39 @@
...
+        props.topics.forEach(topic => {
+          topic.promptTags.forEach(promptTag => {
+            if(!tags.includes(promptTag)){
+              const newTags = tags
+              newTags.push(promptTag)
+              setTag(newTags)
+            }
+          });
+        });
+      }
```
Comment (kjdchapman) [inline components/Feature/TopicExplorer/index.js:34]: nit: I think these lines are indented one step too much?

```suggestion
    if(props.topics){
      props.topics.forEach(topic => {
        topic.promptTags.forEach(promptTag => {
          if(!tags.includes(promptTag)){
            const newTags = tags
            newTags.push(promptTag)
            setTag(newTags)
          }
        });
      });
    }
```

File: components/Feature/TopicExplorer/index.test.js [hunk 2]
```diff
@@ -34,16 +34,24 @@ describe('TopicExplorer', () => {
       expect(screen.queryByText('topic two')).toBeNull();
     });
 
+    it('shows autocomletion of search terms', () =>{
```
Comment (kjdchapman) [inline components/Feature/TopicExplorer/index.test.js:37]: nit: typo (I've tweaked the phrasing slightly too :speak_no_evil:)
```suggestion
    it('shows auto-complete for search terms', () =>{
```

File: components/Feature/TopicExplorer/index.test.js [hunk 3]
```diff
@@ -34,16 +34,24 @@ describe('TopicExplorer', () => {
...
+      fireEvent.change(screen.getByLabelText('Try searching for keywords like'), {
+        target: { value: 'f' },
+      });
+      expect(screen.getByText('fourth tag')).toBeInTheDocument
+      expect(screen.getByText('fifth tag')).toBeInTheDocument
```
Comment (kjdchapman) [inline components/Feature/TopicExplorer/index.test.js:42]: I'd explicitly test here than the _content_ for these tags _isn't_ in the document - think it's cheap and intention-revealing.

```suggestion
      expect(screen.getByText('fifth tag')).toBeInTheDocument
      expect(screen.queryByText('topic one)).toBeNull
```

File: components/Feature/TopicExplorer/index.js [hunk 4]
```diff
@@ -1,25 +1,39 @@
...
+          if(tag.toLocaleLowerCase() == newSearchTerm){
+            results.push(topic);
+          }
       });
     }
     return results
   }
 
+  useEffect(() => {
+      if(props.topics){
```
Comment (kjdchapman) [inline components/Feature/TopicExplorer/index.js:24]: I think a guard clause here is clearer, though I'm not sure how to write them in javascript idiomatically. Maybe:
```javascript
if (!props.topics) { return }
```

File: components/Feature/TopicExplorer/index.js [hunk 5]
```diff
@@ -38,21 +52,29 @@ const TopicExplorer = (props) => {
...
           className="govuk-input govuk-input--width-20"
           value={searchTerm}
           onChange={onSearchChange}
         />
-
-
+        <datalist id="input-tags">  
+        {
+            tags.sort().map((t,i) => (
+              <option>{t}</option>
```
Comment (kjdchapman) [inline components/Feature/TopicExplorer/index.js:74]: Do you get a React warning for this? I would expect to need to assign [unique keys to each option](https://reactjs.org/docs/lists-and-keys.html#keys), like so:
```suggestion
              <option key={i}>{t}</option>
```

File: components/Feature/TopicExplorer/index.js [hunk 6]
```diff
@@ -1,25 +1,39 @@
...
   }
 
+  useEffect(() => {
+      if(props.topics){
+        props.topics.forEach(topic => {
+          topic.promptTags.forEach(promptTag => {
+            if(!tags.includes(promptTag)){
+              const newTags = tags
+              newTags.push(promptTag)
+              setTag(newTags)
```
Comment (kjdchapman) [inline components/Feature/TopicExplorer/index.js:30]: I have trouble parsing this nested loop, but I'd rather merge it and then play with it as we keep working on it - if the tests are passing, I'm happy! :grin:

**Review by kjdchapman — APPROVED** [review]
Nice one - I was expecting this to be something we couldn't fit in!

I've got suggestions, but they're mainly just typos/nitpicks - feel free to merge this whenever you're happy with it! :rocket:

File: components/Feature/TopicExplorer/index.js (hunk 5 above)
Comment (maysakanoni) [reply components/Feature/TopicExplorer/index.js]: I've updated this as part of the previous commit 😄

File: components/Feature/TopicExplorer/index.js (hunk 4 above)
Comment (maysakanoni) [reply components/Feature/TopicExplorer/index.js]: I've updated this as part of the previous commit 😄

File: components/Feature/TopicExplorer/index.test.js (hunk 3 above)
Comment (maysakanoni) [reply components/Feature/TopicExplorer/index.test.js:42]: I've updated this as part of the previous commit 😄

File: components/Feature/TopicExplorer/index.test.js (hunk 2 above)
Comment (maysakanoni) [reply components/Feature/TopicExplorer/index.test.js]: I've updated this as part of the previous commit 😄

**Pull request merged**